src/commands.rs

```diff
@@ -1,38 +1,78 @@
 use crate::myc::constants::{CapabilityFlags, Command as CommandByte};
 
 #[derive(Debug)]
-pub struct ClientHandshake<'a> {
+pub struct ClientHandshake {
     capabilities: CapabilityFlags,
     maxps: u32,
     collation: u16,
-    username: &'a [u8],
+    pub(crate) db: Option<Vec<u8>>,
+    pub(crate) username: Vec<u8>,
+    pub(crate) auth_response: Vec<u8>,
+    pub(crate) auth_plugin: Vec<u8>,
```

Comment:
Let's continue to use `&'a [u8]`  for these to avoid the cloning.

Reply:
1.  Send Handshake with default plugin
2. Get response ClientHandshake
3. Get auth plugin by username
4. If the user's auth plugin is not matched with Client's auth plugin, will send `AuthSwitch` Requests,


But if the `ClientHandshake` use `&'a [u8]`, we can't burrow `self.writer` as mutable again, because it already burrowed as immutable.

Reply:
Hmm, but in that case you can just drop the `ClientHandshake` before using `self.writer` again? If there's any data you need to keep, then just clone it there, rather than force it for every handshake.